extensions/ql-vscode/src/query-history-info.ts

```diff
@@ -0,0 +1,19 @@
+import { RemoteQueryHistoryItem } from './remote-queries/remote-query-history-item';
+import { VariantAnalysisHistoryItem } from './remote-queries/variant-analysis-history-item';
+import { LocalQueryInfo } from './query-results';
+import { assertNever } from './pure/helpers-pure';
+
+export type QueryHistoryInfo = LocalQueryInfo | RemoteQueryHistoryItem | VariantAnalysisHistoryItem;
+
+export function getRawName(item: QueryHistoryInfo): string {
```

Comment:
We should probably add some unit tests around these functions too.

Reply:
Yes! I'll pull those changes (from our pairing session) to a follow-up PR if that's okay 🧹